### ISSUE TITLE
Upgrade to Superset 3 and customization of superset_config.py

### ISSUE THREAD

**Issue by billantoniadis** [alvindaiyan/aws-superset-cdk#10]
Hi Yan. Thank you for sharing this with the public. Do you have any plans to update this to support Superset 3? It looks like they added support for it here: https://aws-ia-us-east-1.s3.us-east-1.amazonaws.com/cfn-ps-apache-superset/templates/superset-entrypoint-existing-vpc.template.yaml . I was going to take a crack at this myself but wanted to check if this is already in the works.

Also, do you have any advice on how to customize the superset_config.py to support features like dashboard embedding. I got that working on an EC2 instance I built to demo Superset and was hoping this was something that was easy to do with this CDK but don't see a straight-forward way to do this.

**Comment by alvindaiyan** [comment]
Hi @billantoniadis, thanks for you issue, it's always great to know someone is interested. I've see the linked yaml just now,  they updated to use image with tag 'latest', it's not very stable if your ecs restart or scaling. 

` Do you have any plans to update this to support Superset 3? `
- For updating to Superset3, I'd happy to, but not planned. 

`Do you have any advice on how to customize the superset_config.py to support features like dashboard embedding?`
- Have you looked at https://github.com/alvindaiyan/aws-superset-cdk/blob/main/src/superset_dashboard.ts in my repo? Is that something you look for?

**Comment by billantoniadis** [comment]
Hi @alvindaiyan, many thanks for the support and the quick reply. 

You are right on it not being safe to use the 'latest' tag. It's definitely safer to use a specific tag like 'v3.0.0'.

Sorry for the confusion regarding embedding a dashboard. I was referring to embedding superset dashboards within a website (not AWS cloudwatch dashboards), and also tweaking some other superset configurations. The [Superset documentation](https://superset.apache.org/docs/installation/configuring-superset) recommends modifying the superset_config.py file. I was wondering how I would go about doing that?

**Comment by alvindaiyan** [comment]
@billantoniadis for embedding the dashboard, I haven't done that before. But the setup.py can help you get token and access to your superset server from your client webapp.

There are few good tutorial you may want have a look: https://medium.com/@khushbu.adav/embedding-superset-dashboards-in-your-react-application-7f282e3dbd88

**Comment by billantoniadis** [comment]
Hi @alvindaiyan thank you for the reply again and for the link. I think the way to have superset read a custom configuration is to extend the PYTHONPATH to include the location where the superset_config.py will be placed. I was thinking about adding a folder in the EFS folder /app/superset_home (or maybe a separate folder) where I can place a superset_config.py file that can be accessed by all the containers running superset. Thoughts?

Also, I made a clone of this repo and updated to use Superset v3.0.0. If it is helpful, I can do a pull request so you can consider whether it makes sense to add to this repo.

**Comment by alvindaiyan** [comment]
@billantoniadis I'm very happy for you to do a pull request, and let me know if any questions.

**Comment by billantoniadis** [comment]
Thank you @alvindaiyan, I just submitted my pull request. 

As an aside, one thing I noticed while testing is that no matter what password I set on projen deploy command line, the superset admin password is always "admin". For example, if I set the password as:

--parameters superset-cdk-dev:supersetpassword="1q2w3e4r"

It is still being set as "admin" instead of "1q2w3e4r". I think that is a defect that existed prior to my changes.
